src/ngraph/runtime/cpu/cpu_executor.cpp

```diff
@@ -24,32 +24,32 @@ static int GetNumCores()
     const auto ngraph_intra_op_parallelism = std::getenv("NGRAPH_INTRA_OP_PARALLELISM");
     int count = 0;
 
-    if (omp_num_threads && (count = std::atoi(omp_num_threads)))
+    if (omp_num_threads)
     {
-        return count;
+        count = std::atoi(omp_num_threads);
     }
-    else if (ngraph_intra_op_parallelism && (count = std::atoi(ngraph_intra_op_parallelism)))
+    else if (ngraph_intra_op_parallelism)
     {
-        return count;
+        count = std::atoi(ngraph_intra_op_parallelism);
     }
     else
     {
-        count = std::thread::hardware_concurrency() >> 1;
+        count = std::thread::hardware_concurrency() / 2;
     }
-    return count ? count : 1;
+    return count < 1 ? 1 : count;
```

Comment:
i wonder if it makes sense to set `int count = 1`. Then we don't really have to check if it was set or not? We already have a `catch-all` else; so `count` should not be equal to `0`.